src/main/java/org/opensearch/ad/rest/handler/InsightsJobActionHandler.java

```diff
@@ -0,0 +1,476 @@
+                        job.getName(),
+                        job.getSchedule(),
+                        job.getWindowDelay(),
+                        false,
+                        job.getEnabledTime(),
+                        Instant.now(),
+                        Instant.now(),
+                        job.getLockDurationSeconds(),
+                        job.getUser(),
+                        job.getCustomResultIndexOrAlias(),
+                        job.getAnalysisType()
+                    );
+
+                    indexJob(disabledJob, listener, "Insights job stopped successfully");
+
+                } catch (IOException e) {
+                    logger.error("Failed to parse insights job", e);
+                    listener.onFailure(new OpenSearchStatusException("Failed to parse insights job", RestStatus.INTERNAL_SERVER_ERROR));
+                }
+            }, e -> {
+                logger.error("Failed to get insights job", e);
+                listener.onFailure(e);
+            }));
+        }
+    }
+
+    /**
+     * Create a new insights job or re-enable existing disabled job.
+     */
+    private void createOrEnableJob(String frequency, User user, ActionListener<InsightsJobResponse> listener) {
+        GetRequest getRequest = new GetRequest(CommonName.JOB_INDEX).id(ADCommonName.INSIGHTS_JOB_NAME);
+
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            client.get(getRequest, ActionListener.wrap(response -> {
+                if (response.isExists()) {
+                    // Job exists, check if it's already enabled
+                    try (
+                        XContentParser parser = RestHandlerUtils
+                            .createXContentParserFromRegistry(xContentRegistry, response.getSourceAsBytesRef())
+                    ) {
+                        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+                        Job existingJob = Job.parse(parser);
+
+                        if (existingJob.isEnabled()) {
+                            logger.info("Insights job is already running");
+                            listener.onResponse(new InsightsJobResponse("Insights job is already running"));
+                            return;
+                        }
+
+                        // Use current user if provided, otherwise keep existing user (for BWC)
+                        Job enabledJob = new Job(
+                            existingJob.getName(),
+                            createSchedule(frequency),
+                            existingJob.getWindowDelay(),
+                            true,
+                            Instant.now(),
+                            null,
+                            Instant.now(),
+                            existingJob.getLockDurationSeconds(),
+                            user != null ? user : existingJob.getUser(),
```

Comment:
If we allow overwrite user, it would enable malicious user to update a job with. new user and thus cause original user to lose access.